core/services/pipeline/scheduler.go

```diff
@@ -125,7 +125,7 @@ func (s *scheduler) reconstructResults() {
 			result.Error = errors.New(r.Error.String)
 		}
 
-		if !r.Output.Null {
+		if r.Output != nil && !r.Output.Null {
```

Comment:
This nil check is required to avoid a panic when EA responds with "error" key, but I'm not sure why or if it's correct.